source/common/http/filter_manager.cc

```diff
@@ -918,9 +925,18 @@ void FilterManager::sendDirectLocalReply(
           },
           [&](ResponseHeaderMap& response_headers, Code& code, std::string& body,
               absl::string_view& content_type) -> void {
+            /* Direct local reply rewrites happen here since we are not going through the filter
```

Comment:
Should explicitly clarify "must happen here since..."

Reply:
Local replies in the general case do go through the filter chain. However, in this direct reply case (e.g. on downstream timeouts where headers have already been sent), we are not going through the filter chain so we continue to call `local_reply_.rewrite` here.